packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/ChannelIncentivesPayout/ChannelIncentivesPayout.tsx

```diff
@@ -0,0 +1,146 @@
+import { generateJsonPayloadFromPayoutsVector, generateSerializedPayload } from '@joystream/js/content'
+import * as multihash from 'multihashes'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useFormContext } from 'react-hook-form'
+import styled from 'styled-components'
+
+import { useApi } from '@/api/hooks/useApi'
+import { CurrencyName } from '@/app/constants/currency'
+import { FileDropzone } from '@/common/components/FileDropzone'
+import { InlineToggleWrap, InputComponent, Label, ToggleCheckbox, TokenInput } from '@/common/components/forms'
+import { Loading } from '@/common/components/Loading'
+import { Row } from '@/common/components/Modal'
+import { RowGapBlock } from '@/common/components/page/PageContent'
+import { useObservable } from '@/common/hooks/useObservable'
+import {
+  channelPayoutsFileValidator,
+  getChannelPayoutsValidatedFiles,
+} from '@/proposals/modals/AddNewProposal/components/SpecificParameters/ChannelIncentivesPayout/helpers'
+
+const MAX_FILE_SIZE = 3 * 1024 * 1024
+
+export const ChannelIncentivesPayout = () => {
+  const { api } = useApi()
+  const [isProcessingFile, setIsProcessingFile] = useState<boolean>(false)
+  const { setValue } = useFormContext()
+  const expectedDataSizeFee = useObservable(() => api?.query.storage.dataObjectPerMegabyteFee(), [api?.isConnected])
+  const expectedDataObjectStateBloatBond = useObservable(
+    () => api?.query.storage.dataObjectStateBloatBondValue(),
+    [api?.isConnected]
+  )
+
+  useEffect(() => {
+    if (expectedDataSizeFee && expectedDataObjectStateBloatBond) {
+      setValue('channelIncentivesPayout.expectedDataSizeFee', expectedDataSizeFee)
+      setValue('channelIncentivesPayout.expectedDataObjectStateBloatBond', expectedDataObjectStateBloatBond)
```

Comment:
It's nice to have the `objectCreationParamsContent` and the `objectCreationParamsSize` validated separately, but to keep `getSpecificParameters` simple the payload object could be build here:
 ```suggestion
      setValue('channelIncentivesPayout.payload', {
        uploaderAccount: creator.controllerAccount,
        objectCreationParams: { size_: objectCreationParamsSize, ipfsContentId: objectCreationParamsContent },
        expectedDataSizeFee: expectedDataSizeFee,
        expectedDataObjectStateBloatBond: expectedDataObjectStateBloatBond,
      })
```

(with `const [objectCreationParamsSize, objectCreationParamsContent] = watch(['objectCreationParamsSize', 'objectCreationParamsContent'])`)

WDYT?

Reply:
It is kinda validated separately. Both of them are validated at the time of upload. I choose that way because FileDropzone has a nicer validation UI than just plain red text. 

So at the time of setting values both size and params content are validated.

Reply:
I got that but what I'm proposing is to keep these separate validations but also to have `channelIncentivesPayout.payload` (not validated) but set (has a react hook form data) inside the `useEffect` in  `ChannelIncentivesPayout` instead of inside `getSpecificParameters`